### PR TITLE
[H01] ETH could get trapped in the protocol

### PR DESCRIPTION
**Pull request by haythem96**
# [H01] ETH could get trapped in the protocol

## High Level Description

To fix H01 bug, and for more simplicity, this PR completely remove ETH usage in `Controller.sol` by:
- changing `operate` to non-payable function
- updating `_call()` to accept only one argument: CallArgs

The `FlashUnwrap.sol` contract is only made for call action testing purpose without controller payable function.
### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [x] Is your code up to date with the spec? 
- [x] Have you added your tests to the testing doc?

### CODE REVIEW

**Comment by haythem96** [comment]
`WETH.withdraw` fail when called from the Callee, is it because it is getting called from a proxy and WETH withdraw function is `.transfer` ?

**Review by antoncoding — COMMENTED** [review]
This PR doesn't fix the problem if user call `operate` with extra eth attached. 
I thought our conclusion was to remove payable as a property for the operate function?

**Comment by haythem96** [comment]
> This PR doesn't fix the problem if user call `operate` with extra eth attached.
> I thought our conclusion was to remove payable as a property for the operate function?

Yeah the operate function is not payable function.

**Comment by antoncoding** [comment]
> > This PR doesn't fix the problem if user call `operate` with extra eth attached.
> > I thought our conclusion was to remove payable as a property for the operate function?
> 
> Yeah the operate function is not payable function.

Just saw the other PR. Maybe now we remove the flashWrap contract and rename [this PR](https://github.com/opynfinance/GammaProtocol/pull/307/files) to H01 fix?

**Comment by haythem96** [comment]
> > > This PR doesn't fix the problem if user call `operate` with extra eth attached.
> > > I thought our conclusion was to remove payable as a property for the operate function?
> > 
> > 
> > Yeah the operate function is not payable function.
> 
> Just saw the other PR. Maybe now we remove the flashWrap contract and rename [this PR](https://github.com/opynfinance/GammaProtocol/pull/307/files) to H01 fix?

I think this one is meant for fixing the bug, the other PR have additional changes to the Call action.

**Review by antoncoding — APPROVED** [review]

File: test/unit-tests/flashUnwrap.test.ts [hunk 1]
```diff
@@ -96,71 +96,95 @@ contract('Controller', ([owner, accountOwner1]) => {
...
+        senderWethBalanceBefore.minus(senderWethBalanceAfter).toString(),
+        amountToUnwrap.toString(),
         'sender WETH balance mismatch',
       )
     })
-    it('should revert if total msg.value and action.value mismath', async () => {
-      const amountToWrap = new BigNumber(web3.utils.toWei('1', 'ether'))
-      const actionArgs = [
+
+    it('should revert sending data with different lenth than the required one', async () => {
```
Comment (aparnakr) [inline test/unit-tests/flashUnwrap.test.ts:156]:
```suggestion
    it('should revert sending data with different length than the required one', async () => {
```

**Review by aparnakr — APPROVED** [review]

**Pull request merged**